### PR TITLE
Include opamp bridge headers in configmap generation

### PR DESCRIPTION
**Pull request by jdcrouse**
**Description:** 
This PR adds Headers to the opamp bridge spec and passes populated headers into the configmap generation

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-operator/issues/2410

**Testing:** 
added Headers to `TestDesiredConfigMap` in `configmap_test.go` to confirm the field gets set properly

**Documentation:** None

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -1,2 +1,7 @@
 resources:
 - manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (jdcrouse) [inline config/manager/kustomization.yaml:3]: @jaronoff97 👍 or 👎 for this autogenned file

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -379,7 +379,7 @@ spec:
                 - --zap-log-level=info
                 - --zap-time-encoding=rfc3339nano
                 - --feature-gates=+operator.autoinstrumentation.go,+operator.autoinstrumentation.nginx
-                image: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.89.0
+                image: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator
```
Comment (jdcrouse) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:382]: @jaronoff97 same for this autogenned change

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -65,13 +65,13 @@ metadata:
...
     containerImage: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator
-    createdAt: "2023-11-17T13:24:32Z"
+    createdAt: "2023-12-04T20:02:17Z"
     description: Provides the OpenTelemetry components, including the Collector
     operators.operatorframework.io/builder: operator-sdk-v1.29.0
     operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
     repository: github.com/open-telemetry/opentelemetry-operator
     support: OpenTelemetry Community
-  name: opentelemetry-operator.v0.89.0
+  name: opentelemetry-operator.v0.89.0-30-g8a916dd
```
Comment (jaronoff97) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml]: you'll want to remove the suffix here (i.e. leave `opentelemetry-operator.v0.89.0`)

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (jaronoff97) [reply config/manager/kustomization.yaml:3]: yeah remove this change :)

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (jaronoff97) [reply bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:382]: yeah you'll want to keep this version in

File: apis/v1alpha1/opampbridge_types.go [hunk 4]
```diff
@@ -24,6 +24,10 @@ type OpAMPBridgeSpec struct {
 	// OpAMP backend Server endpoint
 	// +required
 	Endpoint string `json:"endpoint"`
+	// Headers is an optional map of headers to use when connecting to the OpAMP Server,
+	// typically used to set access tokens or other authorization headers.
+	// +optional
+	Headers map[string]string `json:"headers,omitempty"`
```
[the remark this answers is not in the document]
Comment (jaronoff97) [reply apis/v1alpha1/opampbridge_types.go:30]: yeah we already have the ability to mount a secret on the environment ([here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1alpha1/opampbridge_types.go#L77-L79)) but we can't use those in the bridge's configuration yet. I _think_ someone would be able to do that after this change, though im not sure the environment variable will be bound in the bridge. Definitely something to be done in a followup.

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**